catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/TestCswQueryResponseTransformer.java

```diff
@@ -349,6 +358,51 @@ public void testMarshalAcknowledgement()
         assertThat(jaxB.getValue(), is(instanceOf(GetRecordsType.class)));
     }
 
+    @Test
+    public void verifyResultOrderIsMaintained() throws CatalogTransformerException, IOException {
```

Comment:
There's nothing in this test that's going to simulate the `Future`s completing out of order.